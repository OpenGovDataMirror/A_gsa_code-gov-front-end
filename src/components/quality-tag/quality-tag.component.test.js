import React from 'react'
import { shallow } from 'enzyme'
import QualityTag from 'components/quality-tag/quality-tag.component'

const props = {
  score: 8.9
}

let wrapper
let instance
describe('components - QualityTag', () => {
  beforeEach(() => {
    wrapper = shallow(<QualityTag {...props} />)
    instance = wrapper.instance()
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render a green quality tag if high data quality score', () => {
      wrapper.setProps({ score: 8.9 })
      expect(wrapper.find('div').prop('className')).toBe(
        'height-4 pin-right width-6 radius-lg display-flex flex-column flex-justify-center margin-top-205 margin-right-3 bg-success-dark'
      )
    })

    it('should render a yellow/brown quality tag if medium data quality score', () => {
      wrapper.setProps({ score: 5.2 })
      expect(wrapper.find('div').prop('className')).toBe(
        'height-4 pin-right width-6 radius-lg display-flex flex-column flex-justify-center margin-top-205 margin-right-3 bg-warning-darker'
      )
    })

    it('should render a red quality tag if low data quality score', () => {
      wrapper.setProps({ score: 2.7 })
      expect(wrapper.find('div').prop('className')).toBe(
        'height-4 pin-right width-6 radius-lg display-flex flex-column flex-justify-center margin-top-205 margin-right-3 bg-error-dark'
      )
    })
  })
})
