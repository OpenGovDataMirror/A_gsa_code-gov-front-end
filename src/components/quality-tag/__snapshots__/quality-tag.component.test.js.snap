// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - QualityTag render should render correctly 1`] = `
<div
  aria-label="high data quality score"
  className="height-4 pin-right width-6 radius-lg display-flex flex-column flex-justify-center margin-top-205 margin-right-3 bg-success-dark"
>
  <span
    className="text-center font-body-3xs text-white"
  >
    8.9
  </span>
</div>
`;
